firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -215,6 +239,23 @@ private void ensureClientConfigured() {
     }
   }
 
+  private FirebaseFirestoreSettings mergeEmulatorSettings(
+      @NonNull FirebaseFirestoreSettings settings,
+      @NonNull EmulatedServiceSettings emulatorSettings) {
+
+    if (!FirebaseFirestoreSettings.DEFAULT_HOST.equals(settings.getHost())) {
+      throw new IllegalStateException(
+          "Cannot change the Firestore host through FirebaseFirestoreSettings when "
```

Comment:
What about:

"Cannot specify the host in FirebaseFirestoreSettings when EmulatedServiceSettings is provided."

It conveys the same information ... in shorter.